### ISSUE TITLE
Archive succeeded, exit status 1 with error, no obvious reason

### ISSUE THREAD

**Issue by ApoorvKhatreja** [fastlane/fastlane#13676]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

`build_app` ends with "Archive Succeeded" but then error with "Exit status: 1". I tried capturing the output of `bundle exec fastlane beta` using the `--capture_output` flag, but I got nothing on my clipboard. I'm copy pasting the last few lines of the output, but I see no obvious errors here.

##### Complete output when running fastlane, including the stack trace and command used

<details>
  <pre>
▸ Generating 'MyApp.app.dSYM'
▸ Copying /Users/apoorv/Library/Developer/Xcode/DerivedData/MyApp-cldopslihickokfuhwjgczqbidqw/Build/Intermediates.noindex/ArchiveIntermediates/MyApp/BuildProductsPath/Release-iphoneos/MyApp\ Today\ Extension.appex
skipping copy phase strip, binary is code signed: /Users/apoorv/Library/Developer/Xcode/DerivedData/MyApp-cldopslihickokfuhwjgczqbidqw/Build/Intermediates.noindex/ArchiveIntermediates/MyApp/IntermediateBuildFilesPath/UninstalledProducts/iphoneos/MyApp Today Extension.appex/MyApp Today Extension
▸ Running script 'Run Script'
▸ Running script '[CP] Embed Pods Frameworks'
▸ Running script '[CP] Copy Pods Resources'
▸ Touching MyApp.app (in target: MyApp)
▸ Archive Succeeded
ERROR [2018-11-13 13:27:45.00]: Exit status: 1

+-------------+-------------------------+
|           Build environment           |
+-------------+-------------------------+
| xcode_path  | /Applications/Xcode.app |
| gym_version | 2.108.0                 |
| sdk         | iPhoneOS12.1.sdk        |
+-------------+-------------------------+

INFO [2018-11-13 13:27:45.08]: ▸ Build system information
INFO [2018-11-13 13:27:45.08]: ▸ warning: Swift 3 mode has been deprecated and will be removed in a later version of Xcode. Please migrate "SwiftURLMatcher" to Swift 4.2 using "Convert > To Current Swift Syntax…" in the Edit menu. (in target 'SwiftURLMatcher')
INFO [2018-11-13 13:27:45.08]: ▸ Build system information
INFO [2018-11-13 13:27:45.08]: ▸ warning: Swift 3 mode has been deprecated and will be removed in a later version of Xcode. Please migrate "SwiftURLMatcher" to Swift 4.2 using "Convert > To Current Swift Syntax…" in the Edit menu. (in target 'SwiftURLMatcher')
INFO [2018-11-13 13:27:45.08]:
ERROR [2018-11-13 13:27:45.08]: ⬆️  Check out the few lines of raw `xcodebuild` output above for potential hints on how to solve this error
WARN [2018-11-13 13:27:45.08]: 📋  For the complete and more detailed error log, check the full log at:
WARN [2018-11-13 13:27:45.09]: 📋  /Users/apoorv/Library/Logs/gym/MyApp-MyApp.log
INFO [2018-11-13 13:27:45.09]:
ERROR [2018-11-13 13:27:45.09]: Looks like fastlane ran into a build/archive error with your project
ERROR [2018-11-13 13:27:45.09]: It's hard to tell what's causing the error, so we wrote some guides on how
ERROR [2018-11-13 13:27:45.09]: to troubleshoot build and signing issues: https://docs.fastlane.tools/codesigning/getting-started/
ERROR [2018-11-13 13:27:45.09]: Before submitting an issue on GitHub, please follow the guide above and make
ERROR [2018-11-13 13:27:45.09]: sure your project is set up correctly.
ERROR [2018-11-13 13:27:45.09]: fastlane uses `xcodebuild` commands to generate your binary, you can see the
ERROR [2018-11-13 13:27:45.09]: the full commands printed out in yellow in the above log.
ERROR [2018-11-13 13:27:45.10]: Make sure to inspect the output above, as usually you'll find more error information there
INFO [2018-11-13 13:27:45.10]:
WARN [2018-11-13 13:27:45.10]: Lane Context:
INFO [2018-11-13 13:27:45.10]: {:DEFAULT_PLATFORM=>:ios, :PLATFORM_NAME=>:ios, :LANE_NAME=>"ios beta"}
ERROR [2018-11-13 13:27:45.10]: Error building the application - see the log above
INFO [2018-11-13 13:27:45.11]: Successfully generated documentation at path '/Users/apoorv/Code/MyApp-iphone/fastlane/README.md'

+------+------------------+-------------+
|           fastlane summary            |
+------+------------------+-------------+
| Step | Action           | Time (in s) |
+------+------------------+-------------+
| 1    | default_platform | 0           |
| 💥   | build_app        | 708         |
+------+------------------+-------------+

ERROR [2018-11-13 13:27:45.14]: fastlane finished with errors
bundler: failed to load command: fastlane (/Users/apoorv/.rbenv/versions/2.5.1/bin/fastlane)
FastlaneCore::Interface::FastlaneBuildFailure: [!] Error building the application - see the log above
  /Users/apoorv/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.108.0/fastlane_core/lib/fastlane_core/ui/interface.rb:163:in `build_failure!'
  /Users/apoorv/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.108.0/fastlane_core/lib/fastlane_core/ui/ui.rb:17:in `method_missing'
  /Users/apoorv/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.108.0/gym/lib/gym/error_handler.rb:73:in `handle_build_error'
  /Users/apoorv/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.108.0/gym/lib/gym/runner.rb:97:in `block in build_app'
  /Users/apoorv/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.108.0/fastlane_core/lib/fastlane_core/command_executor.rb:89:in `execute'
  /Users/apoorv/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.108.0/gym/lib/gym/runner.rb:93:in `build_app'
  /Users/apoorv/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.108.0/gym/lib/gym/runner.rb:18:in `run'
  /Users/apoorv/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.108.0/gym/lib/gym/manager.rb:19:in `work'
  /Users/apoorv/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.108.0/fastlane/lib/fastlane/actions/build_ios_app.rb:64:in `run'
  /Users/apoorv/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.108.0/fastlane/lib/fastlane/runner.rb:253:in `block (2 levels) in execute_action'
  /Users/apoorv/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.108.0/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
  /Users/apoorv/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.108.0/fastlane/lib/fastlane/runner.rb:232:in `block in execute_action'
  /Users/apoorv/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.108.0/fastlane/lib/fastlane/runner.rb:227:in `chdir'
  /Users/apoorv/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.108.0/fastlane/lib/fastlane/runner.rb:227:in `execute_action'
  /Users/apoorv/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.108.0/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
  /Users/apoorv/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.108.0/fastlane/lib/fastlane/fast_file.rb:149:in `method_missing'
  Fastfile:21:in `block (2 levels) in parsing_binding'
  /Users/apoorv/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.108.0/fastlane/lib/fastlane/lane.rb:33:in `call'
  /Users/apoorv/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.108.0/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
  /Users/apoorv/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.108.0/fastlane/lib/fastlane/runner.rb:45:in `chdir'
  /Users/apoorv/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.108.0/fastlane/lib/fastlane/runner.rb:45:in `execute'
  /Users/apoorv/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.108.0/fastlane/lib/fastlane/lane_manager.rb:56:in `cruise_lane'
  /Users/apoorv/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.108.0/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
  /Users/apoorv/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.108.0/fastlane/lib/fastlane/commands_generator.rb:108:in `block (2 levels) in run'
  /Users/apoorv/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
  /Users/apoorv/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
  /Users/apoorv/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
  /Users/apoorv/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.108.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:76:in `run!'
  /Users/apoorv/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
  /Users/apoorv/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.108.0/fastlane/lib/fastlane/commands_generator.rb:333:in `run'
  /Users/apoorv/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.108.0/fastlane/lib/fastlane/commands_generator.rb:41:in `start'
  /Users/apoorv/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.108.0/fastlane/lib/fastlane/cli_tools_distributor.rb:108:in `take_off'
  /Users/apoorv/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.108.0/bin/fastlane:23:in `<top (required)>'
  /Users/apoorv/.rbenv/versions/2.5.1/bin/fastlane:23:in `load'
  /Users/apoorv/.rbenv/versions/2.5.1/bin/fastlane:23:in `<top (required)>'
</pre>
</details>

### Environment

<details>
  <pre>
<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.13.6                                     |
| Ruby                        | 2.5.1                                       |
| Bundler?                    | true                                        |
| Git                         | git version 2.17.2 (Apple Git-113)          |
| Installation Source         | ~/.rbenv/versions/2.5.1/bin/fastlane        |
| Host                        | Mac OS X 10.13.6 (17G3025)                  |
| Ruby Lib Dir                | ~/.rbenv/versions/2.5.1/lib                 |
| OpenSSL Version             | OpenSSL 1.0.2p  14 Aug 2018                 |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 10.1                                        |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# This file contains the fastlane.tools configuration
# You can find the documentation at https://docs.fastlane.tools
#
# For a list of all available actions, check out
#
#     https://docs.fastlane.tools/actions
#
# For a list of all available plugins, check out
#
#     https://docs.fastlane.tools/plugins/available-plugins
#

# Uncomment the line if you want fastlane to automatically update itself
# update_fastlane

default_platform(:ios)

platform :ios do
  desc "Push a new beta build to TestFlight"
  lane :beta do
    build_app(workspace: "MyApp.xcworkspace", scheme: "MyApp")
  end
end

```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
app_identifier("com.MyApp.MyAppMobile") # The bundle identifier of your app
apple_id("apoorv@MyApp.com") # Your Apple email address

itc_team_id("REDACTED") # App Store Connect Team ID
team_id("REDACTED") # Developer Portal Team ID

# For more information about the Appfile, see:
#     https://docs.fastlane.tools/advanced/#appfile

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.108.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                           | Version      |
| ----------------------------- | ------------ |
| did_you_mean                  | 1.2.0        |
| bundler                       | 1.17.1       |
| CFPropertyList                | 3.0.0        |
| concurrent-ruby               | 1.1.3        |
| i18n                          | 0.9.5        |
| minitest                      | 5.11.3       |
| thread_safe                   | 0.3.6        |
| tzinfo                        | 1.2.5        |
| activesupport                 | 4.2.10       |
| public_suffix                 | 2.0.5        |
| addressable                   | 2.5.2        |
| atomos                        | 0.1.3        |
| babosa                        | 1.0.2        |
| claide                        | 1.0.2        |
| fuzzy_match                   | 2.0.4        |
| nap                           | 1.1.0        |
| cocoapods-core                | 1.5.3        |
| cocoapods-deintegrate         | 1.0.2        |
| cocoapods-downloader          | 1.2.2        |
| cocoapods-plugins             | 1.0.0        |
| cocoapods-search              | 1.0.0        |
| cocoapods-stats               | 1.0.0        |
| netrc                         | 0.11.0       |
| cocoapods-trunk               | 1.3.1        |
| cocoapods-try                 | 1.1.0        |
| colored2                      | 3.1.2        |
| escape                        | 0.0.4        |
| fourflusher                   | 2.0.1        |
| gh_inspector                  | 1.1.3        |
| molinillo                     | 0.6.6        |
| ruby-macho                    | 1.3.1        |
| nanaimo                       | 0.2.6        |
| xcodeproj                     | 1.7.0        |
| cocoapods                     | 1.5.3        |
| cocoapods-prune-localizations | 0.3.1        |
| colored                       | 1.2          |
| highline                      | 1.7.10       |
| commander-fastlane            | 4.4.6        |
| declarative                   | 0.0.10       |
| declarative-option            | 0.1.0        |
| unf_ext                       | 0.0.7.5      |
| unf                           | 0.1.4        |
| domain_name                   | 0.5.20180417 |
| dotenv                        | 2.5.0        |
| emoji_regex                   | 0.1.1        |
| excon                         | 0.62.0       |
| multipart-post                | 2.0.0        |
| faraday                       | 0.15.3       |
| http-cookie                   | 1.0.3        |
| faraday-cookie_jar            | 0.0.6        |
| faraday_middleware            | 0.12.2       |
| fastimage                     | 2.1.4        |
| jwt                           | 2.1.0        |
| memoist                       | 0.16.0       |
| multi_json                    | 1.13.1       |
| os                            | 1.0.0        |
| signet                        | 0.11.0       |
| googleauth                    | 0.6.7        |
| httpclient                    | 2.8.3        |
| mime-types-data               | 3.2018.0812  |
| mime-types                    | 3.2.2        |
| uber                          | 0.1.0        |
| representable                 | 3.0.4        |
| retriable                     | 3.1.2        |
| google-api-client             | 0.23.9       |
| json                          | 2.1.0        |
| mini_magick                   | 4.5.1        |
| multi_xml                     | 0.6.0        |
| plist                         | 3.4.0        |
| rubyzip                       | 1.2.2        |
| security                      | 0.1.3        |
| naturally                     | 2.2.0        |
| simctl                        | 1.6.5        |
| slack-notifier                | 2.3.2        |
| terminal-notifier             | 1.8.0        |
| unicode-display_width         | 1.4.0        |
| terminal-table                | 1.8.0        |
| tty-screen                    | 0.6.5        |
| tty-cursor                    | 0.6.0        |
| tty-spinner                   | 0.8.0        |
| word_wrap                     | 1.0.0        |
| rouge                         | 2.0.7        |
| xcpretty                      | 0.3.0        |
| xcpretty-travis-formatter     | 1.0.0        |
</details>

*generated on:* **2018-11-13**
</details>
</pre>
</details>

I redacted some sensitive details here.

**Comment by ApoorvKhatreja** [comment]
@fastlane-bot I have, although it isn't apparent that this issue is related to code signing.

**Comment by ApoorvKhatreja** [comment]
I was able to figure this out. Apparently `gym` was failing because it did not have write permissions for the directory, where it looks like build logs are written to.

`~/Library/Logs/gym`

I was able to chmod that directory to set the appropriate permissions, but IMO this should not be an error that causes the build to fail and fastlane to stop. Just my 2 cents. 

Also, the statement in the logs where I found that the permissions were messed up was formatted like a warning, and not an error, which made things more confusing, since I wasn't seeing any errors and was really confused by the exit with error.
